module/zfs/spa_config.c

```diff
@@ -143,38 +146,36 @@ spa_config_load(void)
 	if (buf != NULL)
 		kmem_free(buf, fsize);
 
-	kobj_close_file(file);
+	zfs_file_close(fp);
 }
 
 static int
 spa_config_remove(spa_config_dirent_t *dp)
 {
-#if defined(__linux__) && defined(_KERNEL)
-	int error, flags = FWRITE | FTRUNC;
-	uio_seg_t seg = UIO_SYSSPACE;
-	vnode_t *vp;
-
-	error = vn_open(dp->scd_path, seg, flags, 0644, &vp, 0, 0);
-	if (error == 0) {
-		(void) VOP_FSYNC(vp, FSYNC, kcred, NULL);
-		(void) VOP_CLOSE(vp, 0, 1, 0, kcred, NULL);
-	}
+	int error = 0;
 
+	if (zfs_file_unlink(dp->scd_path)) {
```

Comment:
A comment here is probably warranted.  Additionally, let's play it safe and only fallback to the truncate behavior on an `EOPNOTSUPP` error.   How about something like:

```
/*
 * Remove the cache file.  If zfs_file_unlink() in not supported by the
 * platform fallback to truncating the file which is functionally equivalent.
 */
```